qiskit/providers/models/pulsedefaults.py

```diff
@@ -104,34 +110,326 @@ def __init__(self, name, **kwargs):
+        """
+        Validate and reformat transport layer inputs to initialize this.
+
+        Args:
+            qubit_freq_est (list[number]): Estimated qubit frequencies in GHz.
+            meas_freq_est (list[number]): Estimated measurement cavity frequencies
+                in GHz.
+            buffer (int): Default buffer time (in units of dt) between pulses.
+            pulse_library (list[PulseLibraryItem]): Pulse name and sample definitions.
+            cmd_def (list[Command]): Operation name and definition in terms of Commands.
+            **kwargs (Dict[str, Any]]): Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self._qubit_freq_est_ghz = qubit_freq_est
+        self._meas_freq_est_ghz = meas_freq_est
+        self._qubit_freq_est_hz = [freq * 1e9 for freq in qubit_freq_est]
+        self._meas_freq_est_hz = [freq * 1e9 for freq in meas_freq_est]
+        # TODO: These should be massaged for the user
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
 
-        super().__init__(**kwargs)
+        # The processed and reformatted circuit operation definitions
+        self._ops_def = defaultdict(dict)
+        # A backwards mapping from qubit to supported operation
+        self._qubit_ops = defaultdict(list)
+        # To enable pulse replacement, track where each pulse item is used
+        self.__pulse_library_usage = defaultdict(list)
+
+        # Build the above dictionaries from pulse_library and cmd_def
+        self.converter = QobjToInstructionConverter(pulse_library, buffer)
+        for op in cmd_def:
+            qubits = _to_tuple(op.qubits)
+            self._qubit_ops[qubits].append(op.name)
+            pulse_insts = []
+            for inst in op.sequence:
+                if inst.name not in ['pv', 'fc', 'acquire', 'snapshot']:
+                    self.__pulse_library_usage[inst.name].append(op)
+                pulse_insts.append(self.converter(inst))
+            self._ops_def[op.name][qubits] = ParameterizedSchedule(*pulse_insts, name=op.name)
+
+    @property
+    def qubit_freq_est(self):
+        """
+        Return the estimated resonant frequency for the given qubit in Hz.
+
+        Returns:
+            list[float]: The frequency of the qubit resonance in Hz.
+        """
+        warnings.warn("The qubit frequency estimation was previously in GHz, and now is in Hz.")
+        return self._qubit_freq_est_hz
+
+    @property
+    def meas_freq_est(self):
+        """
+        Return the estimated measurement stimulus frequency to readout from the given qubit.
+
+        Returns:
+            list[float]: The measurement stimulus frequency in Hz.
+        """
+        warnings.warn("The measurement frequency estimation was previously in GHz, "
+                      "and now is in Hz.")
+        return self._meas_freq_est_hz
+
+    def replace_pulse(self, name, samples):
```

Comment:
Does this deal with refactoring the schedule so that the alignments are the same? Eg. if we use this to replace the pi pulse in our definitions for the CNOT will it move the CR pulses appropriately?

Reply:
Ah, it was supposed to, but I'm realizing now that the way it's written will not work, because the default schedules specify exact times to play pulses. I'll have to fix that. Thank you!! 🙏

Reply:
Moving this to a separate PR